src/Message/Envelope.php

```diff
@@ -48,15 +37,12 @@ public function getMetadata(): array
             [
                 EnvelopeInterface::ENVELOPE_STACK_KEY => array_merge(
                     $this->message->getMetadata()[EnvelopeInterface::ENVELOPE_STACK_KEY] ?? [],
-                    [self::class],
+                    [static::class],
                 ),
             ],
             $this->getEnvelopeMetadata(),
         );
     }
 
-    public function getEnvelopeMetadata(): array
-    {
-        return [];
-    }
+    abstract protected function getEnvelopeMetadata(): array;
```

Comment:
it's good to keep default empty array

Reply:
If metadata is empty - then there is no reason to create an envelope. So, every envelope should have its very own set of metadata attributes.